.github/workflows/release.yml

```diff
@@ -1,28 +1,42 @@
 ---
-name: release
-
-permissions:
-  contents: read
+name: Release
 
 on:
   workflow_dispatch:
     inputs:
-      branch_specifier:
-        description: The branch to release ex. main or 0.6.
+      ref:
+        description: 'Branch or tag ref to run the workflow on'
         required: true
         default: "main"
-        type: string
-
+      version:
+        description: 'The version to release (e.g. 1.2.3). This workflow will automatically perform the required version bumps'
+        required: true
       dry_run:
         description: If set, run a dry-run release
         default: false
         type: boolean
 
+permissions:
+  contents: read
+
 jobs:
+  validate-tag:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Validate tag does not exist on current commit
+        uses: ./.github/workflows/validate-tag
+        with:
+          tag: ${{ env.RELEASE_VERSION }}
+
```

Comment:
Here we should probably do an extra verification step to ensure that the maven project version is the one we expect from the parameters.

Reply:
Ref: https://github.com/elastic/ecs-logging-java/pull/226/commits/16f61a6bde7cecd13553558e4b12a5ffce975cb6